docs/web/content/development/recipes/recipe_tutorial.md

```diff
@@ -0,0 +1,215 @@
+---
+title: Build recipe example
+weight: 4210
+pre: "<i class=\"fa fa-asterisk\" aria-hidden=\"true\"></i> "
+---
+
+In this example we will create a mashling gateway recipe that conditionally invokes a PetStore API.
+The recipe will either register or retrieve pet data. The scenario assumes that only pets with a category name of `dog` or `cat` are supported. If an attempt is made to register any other kind of pet the gateway responds with an error message without hitting the PetStore backend.
+
+
+Open an editor and define event triggers like the following:
+
+```json
+{
+  "mashling_schema": "1.0",
+  "gateway": {
+    "name": "MyProxy",
+    "version": "1.0.0",
+    "description": "This is a simple proxy.",
+    "triggers": [
+      {
+        "name": "MyProxy",
+        "description": "Animals rest trigger - PUT animal details",
+        "type": "github.com/TIBCOSoftware/mashling/ext/flogo/trigger/gorillamuxtrigger",
+        "settings": {
+          "port": "9096"
+        },
+        "handlers": [
+          {
+            "dispatch": "Retrieve",
+            "settings": {
+              "autoIdReply": "false",
+              "method": "GET",
+              "path": "/pets/{petId}",
+              "useReplyHandler": "false"
+            }
+          },
+          {
+            "dispatch": "Register",
+            "settings": {
+              "autoIdReply": "false",
+              "method": "PUT",
+              "path": "/pets",
+              "useReplyHandler": "false"
+            }
+          }
+        ]
+      }
+    ],
+```
+
+There are two handlers created for this HTTP trigger. One for retrieving the pet data through an HTTP GET operation and the other for registering a pet through an HTTP PUT operation. For the details of the handler configuration, see the documentation at https://github.com/TIBCOSoftware/mashling/tree/master/ext/flogo/trigger/gorillamuxtrigger
+
+Add a dispatches section:
+
+```json
+    "dispatches": [
+      {
+        "name": "Retrieve",
+        "routes": [
+          {
+            "steps": [
+              {
+                "service": "GetPet",
+                "input": {
+                  "inputs.pathParams": "${payload.pathParams}"
+                }
+              }
+            ],
+            "responses": [
+              {
+                "if": "JSON.parse(GetPet.response.outputs.code) != 200",
```

Comment:
I don't know what the output of the Flogo flow is like for these calls but the JSON.parse() wrapper for the code might not be necessary. It looks like the PutPet conditional statements don't use it.